library/src/main/java/com/dslplatform/json/JsonWriter.java

```diff
@@ -417,11 +417,7 @@ public final void writeAscii(final byte[] buf) {
 		if (position + len >= buffer.length) {
 			enlargeOrFlush(position, len);
 		}
-		final int p = position;
-		final byte[] _result = buffer;
-		for (int i = 0; i < buf.length; i++) {
-			_result[p + i] = buf[i];
-		}
+		System.arraycopy(buf, 0, buffer, position, len);
```

Comment:
Indeed, for small arrays manual coping is faster. For arrays greater 20 bytes it becomes less efficient (but it also depends on machine architecture and JVM version).
```
Input array size 8 bytes:
Benchmark                       Mode  Cnt         Score        Error  Units
BenchTest.writeAscii           thrpt    5  27112980.888 ± 457844.926  ops/s
BenchTest.writeAsciiOptimized  thrpt    5  26028083.793 ± 354472.072  ops/s

Input array size 16 bytes:
Benchmark                       Mode  Cnt         Score        Error  Units
BenchTest.writeAscii           thrpt    5  25680666.334 ± 1849708.424  ops/s
BenchTest.writeAsciiOptimized  thrpt    5  25303425.119 ± 1282369.591  ops/s

Input array size 24 bytes:
Benchmark                       Mode  Cnt         Score        Error  Units
BenchTest.writeAscii           thrpt    5  23191566.449 ± 1368988.544  ops/s
BenchTest.writeAsciiOptimized  thrpt    5  25558813.181 ± 2048939.806  ops/s

Input array size 32 bytes:
Benchmark                       Mode  Cnt         Score        Error  Units
BenchTest.writeAscii           thrpt    5  21894338.881 ± 1608116.034  ops/s
BenchTest.writeAsciiOptimized  thrpt    5  25691447.807 ±  608058.739  ops/s

Input array size 40 bytes:
Benchmark                       Mode  Cnt         Score        Error  Units
BenchTest.writeAscii           thrpt    5  21076890.402 ± 427649.169  ops/s
BenchTest.writeAsciiOptimized  thrpt    5  25894519.090 ± 544010.094  ops/s
```

Reply:
I've rewrote benchmark to make sure. Here is gist of bench: https://gist.github.com/varahash/94d51ba109724ce83bb653eb8d756615
So it seems `System.arraycopy` is faster
Note: number in benchmark name means length of input array
```
Benchmark                          Mode  Cnt          Score          Error  Units
BenchTest.writeAsciiOptimized_04  thrpt   10  192290171.286 ±  2546804.223  ops/s
BenchTest.writeAsciiOptimized_08  thrpt   10  187756061.141 ±  3250500.691  ops/s
BenchTest.writeAsciiOptimized_12  thrpt   10  181318104.113 ±  3187889.132  ops/s
BenchTest.writeAsciiOptimized_16  thrpt   10  139494998.329 ±  1664628.963  ops/s
BenchTest.writeAsciiOptimized_20  thrpt   10  153846964.364 ±  4410827.121  ops/s
BenchTest.writeAsciiOptimized_24  thrpt   10  136970994.793 ±  2415619.763  ops/s
BenchTest.writeAsciiOptimized_28  thrpt   10  142122390.736 ±  2910085.711  ops/s
BenchTest.writeAsciiOptimized_32  thrpt   10  161389766.648 ± 10785565.423  ops/s
BenchTest.writeAsciiOptimized_36  thrpt   10  157999544.768 ±  6489062.043  ops/s
BenchTest.writeAsciiOptimized_40  thrpt   10  152227536.573 ±  5756445.120  ops/s
BenchTest.writeAscii_04           thrpt   10  191075602.289 ±  3880397.430  ops/s
BenchTest.writeAscii_08           thrpt   10  146382254.569 ±  2420333.436  ops/s
BenchTest.writeAscii_12           thrpt   10  119551035.809 ±  1460098.054  ops/s
BenchTest.writeAscii_16           thrpt   10  102816480.367 ±  1767948.989  ops/s
BenchTest.writeAscii_20           thrpt   10   90433811.775 ±  1222619.227  ops/s
BenchTest.writeAscii_24           thrpt   10   80368843.350 ±   923823.610  ops/s
BenchTest.writeAscii_28           thrpt   10   72616706.242 ±  1114834.329  ops/s
BenchTest.writeAscii_32           thrpt   10   65818738.560 ±  1077585.820  ops/s
BenchTest.writeAscii_36           thrpt   10   60287252.512 ±  1448997.142  ops/s
BenchTest.writeAscii_40           thrpt   10   55844046.987 ±   724852.758  ops/s
```

Reply:
Can you test it on: http://github.com/fabienrenaud/java-json-benchmark ?
Also it would be nice to test it on: https://github.com/eishay/jvm-serializers
Using latest java 8 and java 11
If there is no significant changes we can switch to arraycopy

I'll test on my machines so we have different data points

Reply:
Yeah, in overall performance is slightly worse. I'll close this PR. 
Maybe it is worth to put comment there, that this was made intentionally, because IDEA complains about using manual array copy.

Reply:
Sure, add a comment